packages/api/internal/handlers/sandbox_kill.go

```diff
@@ -27,30 +28,69 @@ func (a *APIStore) DeleteSandboxesSandboxID(
 		attribute.String("team.id", teamID.String()),
 	)
 
+	telemetry.ReportEvent(ctx, "killing sandbox")
+
 	sbx, err := a.orchestrator.GetSandbox(sandboxID)
+	if err == nil {
+		if *sbx.TeamID != teamID {
+			errMsg := fmt.Errorf("sandbox '%s' does not belong to team '%s'", sandboxID, teamID.String())
+			telemetry.ReportCriticalError(ctx, errMsg)
+
+			a.sendAPIStoreError(c, http.StatusUnauthorized, fmt.Sprintf("Error killing sandbox - sandbox '%s' does not belong to your team '%s'", sandboxID, teamID.String()))
+
+			return
+		}
+
+		found := a.orchestrator.DeleteInstance(ctx, sandboxID)
+		if !found {
+			a.sendAPIStoreError(c, http.StatusNotFound, fmt.Sprintf("Error killing sandbox - sandbox '%s' was not found", sandboxID))
+
+			return
+		}
+
+		telemetry.ReportEvent(ctx, "deleted sandbox from orchestrator")
+
+		c.Status(http.StatusNoContent)
+		return
+	}
+
+	telemetry.ReportEvent(ctx, "getting paused sandbox from db")
+
+	env, builds, err := a.db.GetSnapshotBuilds(ctx, sandboxID, teamID)
 	if err != nil {
+		telemetry.ReportError(ctx, fmt.Errorf("error when getting paused sandbox from db: %w", err))
 		a.sendAPIStoreError(c, http.StatusNotFound, fmt.Sprintf("Error killing sandbox - sandbox '%s' was not found", sandboxID))
 
 		return
 	}
 
-	sbx.Logger.Debugf("Sandbox kill request received")
+	buildIds := make([]uuid.UUID, len(builds))
+	for i, build := range builds {
+		buildIds[i] = build.ID
+	}
 
-	if *sbx.TeamID != teamID {
-		errMsg := fmt.Errorf("sandbox '%s' does not belong to team '%s'", sandboxID, teamID.String())
+	// delete all builds
+	deleteJobErr := a.templateManager.DeleteBuilds(ctx, buildIds)
+	if deleteJobErr != nil {
+		errMsg := fmt.Errorf("error when deleting env files from storage: %w", deleteJobErr)
 		telemetry.ReportCriticalError(ctx, errMsg)
+	} else {
+		telemetry.ReportEvent(ctx, "deleted env from storage")
+	}
 
-		a.sendAPIStoreError(c, http.StatusUnauthorized, fmt.Sprintf("Error killing sandbox - sandbox '%s' does not belong to your team '%s'", sandboxID, teamID.String()))
+	dbErr := a.db.DeleteEnv(ctx, env.ID)
```

Comment:
I think you should first delete from db. If this fails you will end up in inconsistent state, template is still in db, but you don't have files